tests/test_encoder_decoder_models.py

```diff
@@ -12,35 +12,171 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import tempfile
-import unittest
 
+import pytest
 import torch
-from parameterized import parameterized
 from transformers import AutoModelForSeq2SeqLM, AutoModelForTokenClassification
 
-from peft import LoraConfig, PromptEncoderConfig, TaskType, get_peft_model
-
-from .testing_common import PeftCommonTester, PeftTestConfigManager
+from peft import (
+    AdaLoraConfig,
+    BOFTConfig,
+    BoneConfig,
+    FourierFTConfig,
+    HRAConfig,
+    IA3Config,
+    LoraConfig,
+    OFTConfig,
+    PrefixTuningConfig,
+    PromptEncoderConfig,
+    PromptTuningConfig,
+    TaskType,
+    VBLoRAConfig,
+    VeraConfig,
+    get_peft_model,
+)
+
+from .testing_common import PeftCommonTester
+from .testing_utils import set_init_weights_false
 
 
 PEFT_ENCODER_DECODER_MODELS_TO_TEST = [
     "ybelkada/tiny-random-T5ForConditionalGeneration-calibrated",
     "hf-internal-testing/tiny-random-BartForConditionalGeneration",
 ]
 
-FULL_GRID = {"model_ids": PEFT_ENCODER_DECODER_MODELS_TO_TEST, "task_type": "SEQ_2_SEQ_LM"}
-
-
-class PeftEncoderDecoderModelTester(unittest.TestCase, PeftCommonTester):
-    r"""
-    Test if the PeftModel behaves as expected. This includes:
-    - test if the model has the expected methods
+# Note: Missing from this list are LoKr, LoHa, LN Tuning
```

Comment:
Any particular reason? If so, this would be helpful to add.

Reply:
I think it was just forgotten or not considered important when those methods were added. My goal with the refactors was to replicate as close as possible the existing tests (hence I check the line coverage), modifying the test matrix would counteract that goal (and who knows if these methods pass). I agree though that eventually they should be added.

Reply:
Then this note should also state that they are supposed to be added when there's time. Without that the reader won't benefit from this note IMO